Modules/pyexpat.c

```diff
@@ -1848,7 +1808,127 @@ MODULE_INITFUNC(void)
     MYCONST(XML_CQUANT_REP);
     MYCONST(XML_CQUANT_PLUS);
 #undef MYCONST
+    return 0;
+}
+
+#if XML_COMBINED_VERSION > 19505
+static int
+add_features(PyObject *mod) {
+    PyObject *list = PyList_New(0);
+    if (list == NULL) {
+        return -1;
+    }
 
+    const XML_Feature *features = XML_GetFeatureList();
+    for (size_t i = 0; features[i].feature != XML_FEATURE_END; ++i) {
+        PyObject *item = Py_BuildValue("si", features[i].name,
+                                       features[i].value);
+        if (item == NULL) {
+            goto error;
+        }
+        int ok = PyList_Append(list, item);
+        Py_DECREF(item);
+        if (ok < 0) {
+            goto error;
+        }
+    }
+    if (PyModule_AddObject(mod, "features", list) < 0) {
+        goto error;
+    }
+
+    return 0;
+error:
```

Comment:
Nitpick: I like to add an empty line before an error label for readability.

```suggestion
    }
    return 0;
    
error:
```